frontend/src/app/projects/project-detail/project-volumes/project-volumes.component.ts

```diff
@@ -0,0 +1,123 @@
+  Component,
+  inject,
+  model,
+  OnInit,
+} from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatDialog } from '@angular/material/dialog';
+import { MatIconModule } from '@angular/material/icon';
+import { MatTooltipModule } from '@angular/material/tooltip';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { NgxSkeletonLoaderComponent } from 'ngx-skeleton-loader';
+import { filter, switchMap, take } from 'rxjs';
+import { RelativeTimeComponent } from 'src/app/general/relative-time/relative-time.component';
+import { ConfirmationDialogComponent } from 'src/app/helpers/confirmation-dialog/confirmation-dialog.component';
+import { ProjectsVolumesService, ProjectVolume } from 'src/app/openapi';
+import { ProjectUserService } from 'src/app/projects/project-detail/project-users/service/project-user.service';
+import { ProjectWrapperService } from 'src/app/projects/service/project.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-project-volumes',
+  imports: [
+    MatTooltipModule,
+    MatIconModule,
+    NgxSkeletonLoaderComponent,
+    AsyncPipe,
+    MatButtonModule,
+    RelativeTimeComponent,
+  ],
+  templateUrl: './project-volumes.component.html',
+  styles: `
+    :host {
+      display: block;
+    }
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProjectVolumesComponent implements OnInit {
+  private projectVolumeService = inject(ProjectsVolumesService);
+  public projectWrapperService = inject(ProjectWrapperService);
+  public projectUserService = inject(ProjectUserService);
+  private matDialog = inject(MatDialog);
+
+  projectVolume = model<ProjectVolume | null | undefined>(undefined);
+  volumeCreationInProgress = model<boolean>(false);
+
+  ngOnInit(): void {
+    this.reloadVolumes();
+  }
+
+  reloadVolumes() {
+    this.projectWrapperService.project$
+      .pipe(
+        filter(Boolean),
+        take(1),
+        switchMap((project) =>
+          this.projectVolumeService.getProjectVolume(project.slug),
+        ),
+      )
+      .subscribe((volume) => {
+        this.projectVolume.set(volume);
+      });
+  }
+
+  addVolume() {
+    this.volumeCreationInProgress.set(true);
+    this.projectWrapperService.project$
+      .pipe(
+        filter(Boolean),
+        take(1),
+        switchMap((project) =>
+          this.projectVolumeService.createProjectVolume(project.slug),
+        ),
+      )
+      .subscribe({
+        next: () => {
+          this.volumeCreationInProgress.set(false);
+          this.reloadVolumes();
+        },
+        error: () => {
+          this.volumeCreationInProgress.set(false);
+        },
+      });
+  }
+
+  deleteVolume(projectVolume: ProjectVolume) {
+    this.matDialog
+      .open(ConfirmationDialogComponent, {
+        data: {
+          title: 'Remove project volume',
+          text: 'Do you want to delete the project volume? This will irrevocably delete all files in the workspace.',
```

Comment:
This warning should contain some sort of info about the files that are about to be deleted to prevent accidental deletions

Reply:
We don't really have any information about the files at this stage. It's not trivial to get to the information since the volume is not mounted at this point. We would need to spawn a Pod to see the files, which is to advances for this route. However, I'll include the project name for reference. 

*It might be easier in the future when users can also see the files of shares in the file browser*